plugins/modules/site_workflow_manager.py

```diff
@@ -555,111 +554,147 @@ def get_current_site(self, site):
-                   - site_exists (bool): Indicates whether the site exists.
-                   - dict: Contains information about the existing site. If the
-                           site doesn't exist, this dictionary is empty.
+        - tuple: A tuple containing a boolean indicating whether the site exists and
+                a dictionary containing information about the existing site.
+                The returned tuple includes two elements:
+                - site_exists (bool): Indicates whether the site exists.
+                - dict: Contains information about the existing site. If the
+                        site doesn't exist, this dictionary is empty.
+
         Description:
             Checks the existence of a site in Cisco Catalyst Center by querying the
-          'get_site' function in the 'sites' family. It utilizes the
-          'site_name' parameter from the 'want' attribute to identify the site.
+        'get_site' function in the 'sites' family. It utilizes the
+        'site_name_hierarchy' parameter from the 'want' attribute to identify the site.
         """
         site_exists = False
         current_site = {}
-        response = None
 
         if self.compare_dnac_versions(self.get_ccc_version(), "2.3.5.3") <= 0:
-            site_name = self.want.get("site_name")
-            response = self.get_site(site_name)
+            site_name_hierarchy = self.want.get("site_name_hierarchy")
+            response = self.get_site_v1(site_name_hierarchy)
+
             if not response:
-                self.log("No response received from 'get_site' API for site: {0}".format(site_name), "ERROR")
-                return (site_exists, current_site)
+                self.log("No response received from 'get_site' API for site: {0}".format(site_name_hierarchy), "ERROR")
+                return site_exists, current_site
 
-            response = response.get("response")
-            self.log("Received API response from 'get_site': {0}".format(str(response)), "DEBUG")
+            response_data = response.get("response")
+            self.log("Received API response from 'get_site': {0}".format(str(response_data)), "DEBUG")
 
-            current_site = self.get_current_site(response)
+            current_site = self.get_current_site(response_data)
             if current_site:
                 site_exists = True
-                self.log("Site '{0}' exists in Cisco Catalyst Center".format(site_name), "INFO")
+                self.log("Site '{0}' exists in Cisco Catalyst Center".format(site_name_hierarchy), "INFO")
             else:
-                self.log("No valid site details found for '{0}'".format(site_name), "WARNING")
+                self.log("No valid site details found for '{0}'".format(site_name_hierarchy), "WARNING")
+            return site_exists, current_site
 
-            return (site_exists, current_site)
+        try:
+            bulk_operation = self.config[0].get('site', {}).get('bulk_operation', False)
```

Comment:
Do we support bulk operation for the CatC version below 2.3.5.3?

Reply:
No, Modified code  and Added else condition now